### DISCUSSION TITLE
How to comment out a statement or code block

### DISCUSSION THREAD

**Discussion by lizuncong** [babel/babel#16744] [Q&A]
for example：

```javascript
const test = () => {

  console.log('a')

}
```

comment out as

```javascript
const test = () => {

  //console.log('a')

}

```

I just use the console as an example, it can be other business code. I just want to comment it out, don't want to delete it, because I may need to restore this code later

**Comment by nicolo-ribaudo** [comment]
You could replace it with an empty statement: https://astexplorer.net/#/gist/e27b27e3999747664ba74bdc1305f391/latest

**Comment by lizuncong** [comment]
This will produce extra semicolons @nicolo-ribaudo

**Comment by lizuncong** [comment]
I am currently upgrading a legacy project. Some jsx attributes in the old project, such as those starting with data-, need to be temporarily commented out, but cannot be removed. After the source code upgrade is completed, it will be modified based on the original data- attributes.
```jsx

export function Item() {

  return (
    <div
      className="item"
      data-trace_id={item.report_info.trace_id}
      data-pkgname={item.pkgname}
      data-appname={item.name}
      id="test_item"
    >
    </div>
  );
}

```

comment out as：
```jsx

export function Item() {

  return (
    <div
      className="item"
      //data-trace_id={item.report_info.trace_id}
      //data-pkgname={item.pkgname}
     // data-appname={item.name}
      id="test_item"
    >
    </div>
  );
}

```

**Comment by lizuncong** [comment]
It work well
```javascript

const types = require('@babel/types');

const commentOutPath = (path, state) => {
    const value = path.toString();
    const node = path.node;
    const comments = value.split("\n").map(line => ({
        type: "CommentLine",
        value: " " + line,
        loc: node.loc
    }));

    const emptyStm = types.noop()

    types.addComments(emptyStm, "leading", comments);
    path.insertAfter(emptyStm)
    path.remove();
    // path.replaceWith(emptyStm);
}
```
![image](https://github.com/user-attachments/assets/9b65a929-0dc8-4c22-9296-35686b24a6f6)
